File: flutter/lib/features/recipes/domain/recipe_tag.dart
class RecipeTag {
  final int id;
  final String name;
  final String color;

  const RecipeTag({
    required this.id,
    required this.name,
    this.color = '#6B7280',
  });

  factory RecipeTag.fromJson(Map<String, dynamic> json) {
    return RecipeTag(
      id: json['id'] as int,
      name: json['name'] as String,
      color: json['color'] as String? ?? '#6B7280',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'color': color,
    };
  }
}

File: backend/app/models/recipe_tag.py
from sqlalchemy import ForeignKey, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base


class RecipeTag(Base):
    __tablename__ = "recipe_tags"
    __table_args__ = (
        UniqueConstraint("family_id", "name", name="uq_recipe_tag_family_name"),
    )

    id: Mapped[int] = mapped_column(primary_key=True)
    family_id: Mapped[int] = mapped_column(
        ForeignKey("families.id", ondelete="CASCADE"), index=True
    )
    name: Mapped[str] = mapped_column(String(50))
    color: Mapped[str] = mapped_column(String(7), default="#6B7280")
